src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -2415,6 +2444,24 @@ public override void SetDebuggerStepMode(bool enabled)
             }
         }
 
+        /// <summary>
+        /// Breaks into the debugger.
+        /// </summary>
+        /// <param name="triggerObject">The object that triggered the breakpoint, if there is one.</param>
+        internal override void Break(object triggerObject = null)
+        {
+            base.Break(triggerObject);
+
+            // If the debugger is running and we are not in a breakpoint, trigger an immediate break in the current location
+            using (IEnumerator<CallStackFrame> enumerator = GetCallStack().GetEnumerator())
+            {
+                if (enumerator.MoveNext() && _context._debuggingMode > 0)
+                {
+                    OnSequencePointHit(enumerator.Current.FunctionContext);
```

Comment:
This is interesting, but seems like it should work.  Much better than waiting until next sequence point!

Reply:
Thanks! It works great in my tests so far!